qcfractal/cli/qcfractal_template_generator.py

```diff
@@ -0,0 +1,438 @@
+
+__all__ = ["main"]
+
+# Using backslash on opening triple quotes ('''\) to indicate no new line after from that line return in final
+
+base_script = '''\
+"""
+{MANAGER_NAME_SHORT} Manager Helper
+
+Conditions:
+- {MANAGER_NAME_LONG}
+- Manager running on the head node
+- {CLUSTER_QUEUE_NAME} manager
+
+For additional information about the {MANAGER_URL_TITLE}, please visit this site:
+{MANAGER_URL}
+"""
+
+# Fractal Settings
+# Location of the Fractal Server you are connecting to
+FRACTAL_ADDRESS = "localhost:7777"
+
+# How many cores per node you want your jobs to have access to
+CORES_PER_NODE = 1
+# How much memory per node (in GB) you want your jobs to have access to
+MEMORY_PER_NODE = 1
+# How many tasks per node you want to execute on
+MAX_TASKS_PER_NODE = 1
+# Maximum number of nodes to try and consume
+MAX_NODES = 1
+# Whether or not to claim nodes for exclusive use. We recommend you do, but that's up to you
+NODE_EXCLUSIVITY = True
+
+# Generic Cluster Settings
+# Additional commands to send to the command line (often used as "#SBATCH ..." or '#PBS' headers.)
+# This is a per-node type setting, not task.
+# Don't set memory or cpu or wall clock through this
+SCHEDULER_OPTS = []
+# Additional commands to start each task with. E.g. Activating a conda environment
+TASK_STARTUP_COMMANDS = ''
+
+{CLUSTER_QUEUE_FLAGS}
+
+###################
+
+# QCFractal import
+import qcfractal
+import qcfractal.interface as portal
+
+# Make sure logging is setup correctly
+import tornado.log
+tornado.log.enable_pretty_logging()
+
+{IMPORTS}
+
+# Quick sanity checks
+
+if CORES_PER_NODE < 1 or not isinstance(CORES_PER_NODE, int):
+    raise ValueError("CORES_PER_NODE must be an integer of at least 1")
+if MAX_TASKS_PER_NODE < 1 or not isinstance(MAX_TASKS_PER_NODE, int):
+    raise ValueError("MAX_TASKS_PER_NODE must be an integer of at least 1")
+if MAX_NODES < 1 or not isinstance(MAX_NODES, int):
+    raise ValueError("MAX_NODES must be an integer of at least 1")
+if MEMORY_PER_NODE <= 0:
+    raise ValueError("MEMORY_PER_NODE must be a number > 0")
+{SANITY_CHECKS}
+
+{MANAGER_CLIENT_BUILDER}
+
+# Build a interface to the server
+# If testing, comment this line out and set the first argument to QueueManager to None
+# Otherwise, uncomment this line and set the first arg to this object
+{FRACTAL_TEST_COMMENT}fractal_client = portal.FractalClient(FRACTAL_ADDRESS, verify=False)
+
+# Build a manager
+manager = qcfractal.queue.QueueManager({FRACTAL_TEST_NONE}, {MANAGER_CLIENT}, update_frequency=0.5,
+                                       cores_per_task=CORES_PER_NODE // MAX_TASKS_PER_NODE,
+                                       memory_per_task=MEMORY_PER_NODE // MAX_TASKS_PER_NODE)
+
+# Important for a calm shutdown
+from qcfractal.cli.cli_utils import install_signal_handlers
+install_signal_handlers(manager.loop, manager.stop)
+
+# Start or test the loop. Swap with the .test() and .start() method respectively
+manager.{TEST_OR_START}()
```

Comment:
Can we change this so that we have a variable like `RUNTEST = True` or similar so that a user can setup an entire manager script, test it, then spin up a real manager without having to re-render?

Reply:
Sure! I think that's a great idea.